main.py

```diff
@@ -0,0 +1,57 @@
+from fastapi import Depends, HTTPException, FastAPI
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.post("/author/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    db_user = crud.get_author_by_name(db, name=author.name)
+    if db_user:
+        raise HTTPException(status_code=400, detail="Author already registered.")
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
+    authors = crud.get_authors(db, skip=skip, limit=limit)
+    return authors
+
+
+@app.get("/authors/{author_id}", response_model=schemas.Author)

```

Comment:
it's generally a good practice for endpoints to conclude with a trailing slash (/).